### PR TITLE
Fix Carthage build issue

### PR DESCRIPTION
**Pull request by cntrump**
## Changes in this pull request

Issue fixed:
- Fix Carthage build issue.
- Remove useless Pod files.

### Checklist

- [x] All tests pass. Demo project builds and runs.
- [x] I added tests, an experiment, or detailed why my change isn't tested.
- [x] I added an entry to the `CHANGELOG.md` for any breaking changes, enhancements, or bug fixes.
- [x] I have reviewed the [contributing guide](https://github.com/Instagram/IGListKit/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by cntrump** [comment]
Cartfile
```
github "Instagram/IGListKit" "master"
```

Build 
```
carthage update --platform ios --use-xcframeworks
```

**Review by lorixx — CHANGES_REQUESTED** [review]
https://github.com/Instagram/IGListKit/pull/1531/checks?check_run_id=3666537897

Hi @cntrump looks like all the Unit tests are failing here, can you fix them and update your PR? Thanks

**Comment by cntrump** [comment]
> https://github.com/Instagram/IGListKit/pull/1531/checks?check_run_id=3666537897
> 
> Hi @cntrump looks like all the Unit tests are failing here, can you fix them and update your PR? Thanks

Thanks for review, fixed and tested, https://github.com/cntrump/IGListKit/runs/3695497205?check_suite_focus=true

**Comment by lorixx** [comment]
@cntrump  Fantastic! The CI run looks great and I am very happy that we got this fixed, I will land this PR on Monday, thanks again for your support to IGListKit!

**Comment by lorixx** [comment]
@cntrump One follow-up question, do you know what was the issue before? and why deleting everything under Pods/ would solve the issue? Thanks!

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -62,6 +65,9 @@ jobs:
       - name: Install ruby gems.
         run: bundle install
 
+      - name: Install pods.
+        run: pod install
```
Comment (lorixx) [inline .github/workflows/CI.yml:69]: This makes a lot of sense to me now!

**Review by lorixx — APPROVED** [review]

**Comment by cntrump** [comment]
> @cntrump One follow-up question, do you know what was the issue before? and why deleting everything under Pods/ would solve the issue? Thanks!

Run `carthage build --platform ios --use-xcframeworks`

```
*** xcodebuild output can be found in /var/folders/5_/ymh1f63s1h93dc24x0y5g1rm0000gn/T/carthage-xcodebuild.KNu0Px.log
*** Building scheme "IGListKit" in IGListKitExamples.xcworkspace
```

Scheme "IGListKit" in IGListKitExamples.xcworkspace was built instead of IGListKit.xcodeproj.

`Pods`, `*.xcworkspace` and `*.lock` can be regenerated by running `bundle install` and `pod install`.

**Pull request closed without merging**